packages/idempotency/src/persistence/IdempotencyRecord.ts

```diff
@@ -1,25 +1,32 @@
-import { IdempotencyRecordStatus } from 'types/IdempotencyRecordStatus';
+import { IdempotencyInvalidStatusError } from '../Exceptions';
+import { IdempotencyRecordStatus } from '../types/IdempotencyRecordStatus';
 
 class IdempotencyRecord {
 
   public constructor(public idempotencyKey: string,
     private status: IdempotencyRecordStatus,
     public expiryTimestamp: number | undefined,
```

Comment:
> Aside from the naming/location however, I think we should consider having the client as a class prop. This would give customers the chance to pass their own client with custom configs (i.e. different region or credentials) as well as apply tracing to it.

@dreamorosi  This will be covered in [FR8](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/447#issuecomment-1217966844).